components/infra-proxy-service/server/v1/orgs_test.go

```diff
@@ -0,0 +1,908 @@
+package v1_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+
+	"github.com/chef/automate/api/external/common/query"
+	secrets "github.com/chef/automate/api/external/secrets"
+	authz "github.com/chef/automate/api/interservice/authz/common"
+	request "github.com/chef/automate/api/interservice/infra_proxy/request"
+	infra_proxy "github.com/chef/automate/api/interservice/infra_proxy/service"
+	"github.com/chef/automate/components/infra-proxy-service/constants"
+	"github.com/chef/automate/components/infra-proxy-service/test"
+	"github.com/chef/automate/lib/grpc/grpctest"
+	"github.com/chef/automate/lib/logger"
+)
+
+func TestOrgs(t *testing.T) {
+	ctx := context.Background()
+
+	l, err := logger.NewLogger("text", "debug")
+	require.NoError(t, err, "could not init logger", err)
+
+	migrationConfig, err := test.MigrationConfigIfPGTestsToBeRun(l, "../../storage/postgres/migration/sql")
+	require.NoError(t, err)
+
+	_, serviceRef, conn, close, authzMock := test.SetupInfraProxyService(ctx, t, l, *migrationConfig)
+	secretsMock := serviceRef.Secrets.(*secrets.MockSecretsServiceClient)
+
+	cl := infra_proxy.NewInfraProxyClient(conn)
+
+	t.Helper()
+	defer close()
+
+	serverRes, err := cl.CreateServer(ctx, &request.CreateServer{
+		Name:        "Chef infra server",
+		Description: "Chef infra server",
+		Fqdn:        "domain.com",
+		IpAddress:   "0.0.0.0",
+	})
+	require.NoError(t, err)
+	require.NotNil(t, serverRes)
+
+	secretID := &secrets.Id{
+		Id: "fake id",
+	}
+	newSecret := secrets.Secret{
+		Name: "infra-proxy-service-admin-key",
+		Type: "ssh",
+		Data: []*query.Kv{
+			{Key: "username", Value: "admin"},
+			{Key: "key", Value: "--KEY--"},
+		},
+	}
+	secretWithID := newSecret
+	secretWithID.Id = "fake id"
+
+	t.Run("CreateOrg", func(t *testing.T) {
+		test.ResetState(ctx, t, serviceRef)
+
+		t.Run("when a valid org is submitted", func(t *testing.T) {
```

Comment:
i find it easier to read these tests when they include the expected behavior.
```suggestion
		t.Run("when a valid org is submitted, creates the new org successfully", func(t *testing.T) {
```